### PR TITLE
Tutorial for examples/advanced/busy-bot

### PR DESCRIPTION
**Pull request by shwetalsoni**
Screenshot from 2021-07-22 00-19-51
Screenshot from 2021-07-22 00-20-09
Screenshot from 2021-07-22 00-20-24
Screenshot from 2021-07-22 00-20-35
Screenshot from 2021-07-22 00-20-48
Screenshot from 2021-07-22 00-21-05
Screenshot from 2021-07-22 00-21-22
Screenshot from 2021-07-22 00-21-36
Screenshot from 2021-07-22 00-21-50
Screenshot from 2021-07-22 00-22-03
Screenshot from 2021-07-22 00-22-15

### CODE REVIEW

**Comment by shwetalsoni** [comment]
Tag #891

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+npx ts-node examples/advanced/busy-bot.js
+```
+
+This will generate a QR code. Scan it using **Wechat/Whatsapp** and you are ready to go.
+
+## Building the bot
+
+Let's get started with building **busy-bot** using Wechaty.
+
+### 1. Initialise project
```
Comment (proudofsimin) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: initialize

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+
+Create a new folder called `busy-bot` and move into that directory.
+
+```bash
+mkdir busy-bot
+cd busy-bot
+```
+
+Use the following command to initialise an npm project
```
Comment (proudofsimin) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: initialize

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+You can find more puppet information here: https://wechaty.js.org/docs/puppet-providers/
+
+Now, you are ready to write main code for bot.
+
+### 3. Writing code for bot
+
+Create a new folder called `src` and add a file `busy-bot.js`. You will be writing code here.
+
+Let's import required packages in `busy-bot.js` and initialise the bot by providing it a name and puppet to be used.
```
Comment (proudofsimin) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: initialize

**Review by proudofsimin — APPROVED** [review]

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+---
+
+<!-- MDX import -->
+import Tabs     from '@theme/Tabs'
+import TabItem  from '@theme/TabItem'
+
+[![Powered by Wechaty](https://img.shields.io/badge/Powered%20By-Wechaty-brightgreen.svg)](https://github.com/Wechaty/wechaty)
+[![TypeScript](https://img.shields.io/badge/%3C%2F%3E-TypeScript-blue.svg)](https://www.typescriptlang.org/)
+
+Busy bot sends auto response message for you when you are busy.
```
Comment (Rohitesh-Kumar-Jain) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: it should be message, messages, or maybe message(s) ?

I am leaving upon you to select this

File: docusaurus/docs/examples/advanced/busy-bot.mdx (hunk 2 above)
Comment (Rohitesh-Kumar-Jain) [reply docusaurus/docs/examples/advanced/busy-bot.mdx]: Well this is intersting

Actually @proudofsimin, In India we follow British English in schools and offices. Anyways what English does Wechaty follows?

I think we should go with the type of English followed in China, and I wasn't able to find on the internet what type of English is being followed there 😅 I saw conflicting anserws on quora https://www.quora.com/Which-is-preferred-in-mainland-China-American-English-or-British-English

File: docusaurus/docs/examples/advanced/busy-bot.mdx (hunk 4 above)
Comment (Rohitesh-Kumar-Jain) [reply docusaurus/docs/examples/advanced/busy-bot.mdx]: anyways is should be **a** message, if a single message is being sent

**Comment by shwetalsoni** [comment]
@Rohitesh-Kumar-Jain and @proudofsimin I have applied the suggested fixes.

**Comment by Rohitesh-Kumar-Jain** [comment]
> @Rohitesh-Kumar-Jain and @proudofsimin I have applied the suggested fixes.

you haven't applied my suggested change : message to a message or depending on the context messages if applicable

**Comment by Rohitesh-Kumar-Jain** [comment]
And We were actually yet to decide which English variant we are following, because as per the English (UK one) taught in our schools, you were actually correct

**Comment by shwetalsoni** [comment]
> > @Rohitesh-Kumar-Jain and @proudofsimin I have applied the suggested fixes.
> 
> you haven't applied my suggested change : message to a message or depending on the context messages if applicable

I thought you agreed upon keeping it `message` and to me even it sounds good. 
Actually what is happening here is:

either we can keep it as **Busy bot sends auto-response message for you when you are busy.** or
**Busy bot sends a message for you when you are busy**. or
but we can't change it to `messages` because only one message is being sent.

**Comment by shwetalsoni** [comment]
> And We were actually yet to decide which English variant we are following, because as per the English (UK one) taught in our schools, you were actually correct

Yeah, that is true but I think since from the beginning in this documentation we have been following English(American one), so to maintain the consistency, its better to use it.

**Comment by sbis04** [comment]
Yes @shwetalsoni,  @Rohitesh-Kumar-Jain I believe we are trying to follow **English (US)** for the Wechaty documentation because previously also we have tried to keep that consistency of English (US), so it would be better to stick to it.

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+```
+
+3. If you want to use wecom, you can install wechaty-puppet-wxwork
+
+```bash
+npm install wechaty-puppet-wxwork
+```
+
+You can find more puppet information here: https://wechaty.js.org/docs/puppet-providers/
+
```
Comment (sbis04) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: I think this part can be formatted in a better way, like this (passing on some suggestions that @Rohitesh-Kumar-Jain had given me once 😅):

1. If you want to use **WhatsApp**, install `wechaty-puppet-whatsapp`:

  ```bash
  npm install wechaty-puppet-whatsapp
  ```
2. If you want to use **WeChat**, install `wechaty-puppet-wechat`:

  ```bash
  npm install wechaty-puppet-wechat
  ```

3. If you want to use **Wecom**, install `wechaty-puppet-wxwork`:

  ```bash
  npm install wechaty-puppet-wxwork
  ```

> You can find more information about the puppets [here](https://wechaty.js.org/docs/puppet-providers/).

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+export WECHATY_LOG=verbose
+export WECHATY_PUPPET=wechaty-puppet-wechat
+```
+
+</TabItem>
+<TabItem value="windows">
+
+```bash
+set WECHATY_LOG=verbose
+set WECHATY_PUPPET=wechaty-puppet-wechat
```
Comment (sbis04) [inline docusaurus/docs/examples/advanced/busy-bot.mdx:367]: In these, you are setting the puppet but earlier in the code the bot is initialized like this:

```ts
const bot = new Wechaty({
  name: "busy-bot",
  puppet: "wechaty-puppet-wechat4u"
});
```

So, I guess this overrides the exported puppets and use `wechaty-puppet-wechat4u` by default, then exporting or giving the choice to install other puppets doesn't make sense. Let me know if I am wrong.

**Review by sbis04 — CHANGES_REQUESTED** [review]

File: docusaurus/docs/examples/advanced/busy-bot.mdx (hunk 6 above)
Comment (shwetalsoni) [reply docusaurus/docs/examples/advanced/busy-bot.mdx:367]: @sbis04 yeah I think you are right, what do you think I should do. I mean, now we don't know if busy bot works with other puppets or not. No one has responded to issue #1119. 
So is it better to remove it during initializing bot or while exporting?

File: docusaurus/docs/examples/advanced/busy-bot.mdx (hunk 5 above)
Comment (shwetalsoni) [reply docusaurus/docs/examples/advanced/busy-bot.mdx]: Thanks @sbis04 for suggesting this.

**Comment by Rohitesh-Kumar-Jain** [comment]
> Yes @shwetalsoni, @Rohitesh-Kumar-Jain I believe we are trying to follow **English (US)** for the Wechaty documentation because previously also we have tried to keep that consistency of English (US), so it would be better to stick to it.

Yes let's stick to the US variant, thanks to @proudofsimin for pointing that

**Comment by Rohitesh-Kumar-Jain** [comment]
> > > @Rohitesh-Kumar-Jain and @proudofsimin I have applied the suggested fixes.
> > 
> > 
> > you haven't applied my suggested change : message to a message or depending on the context messages if applicable
> 
> I thought you agreed upon keeping it `message` and to me even it sounds good.
> Actually what is happening here is:
> 
> either we can keep it as **Busy bot sends auto-response message for you when you are busy.** or
> **Busy bot sends a message for you when you are busy**. or
> but we can't change it to `messages` because only one message is being sent.

Well let's change it to Busy bot sends _**an**_ auto-response message for you when you are busy 😅

**Review by Rohitesh-Kumar-Jain — COMMENTED** [review]
I saw that Simin pointed out 3 initialize changes, and you have changed only 2

I Will give my approval once all comments gets resoved

**Comment by shwetalsoni** [comment]
> I saw that Simin pointed out 3 initialize changes, and you have changed only 2
> 
> I Will give my approval once all comments gets resoved

Sorry, I missed it. Now it's done.

File: docusaurus/docs/examples/advanced/busy-bot.mdx (hunk 6 above)
Comment (sbis04) [reply docusaurus/docs/examples/advanced/busy-bot.mdx:367]: Yeah, I think you can just remove the puppet from the initialization of the bot for now.

Comment (shwetalsoni) [reply docusaurus/docs/examples/advanced/busy-bot.mdx:367]: ok thanks @sbis04

**Review by sbis04 — APPROVED** [review]

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 7]
```diff
@@ -0,0 +1,365 @@
+---
+title: Busy Bot
+---
+
+<!-- MDX import -->
+import Tabs     from '@theme/Tabs'
+import TabItem  from '@theme/TabItem'
+
+[![Powered by Wechaty](https://img.shields.io/badge/Powered%20By-Wechaty-brightgreen.svg)](https://github.com/Wechaty/wechaty)
+[![TypeScript](https://img.shields.io/badge/%3C%2F%3E-TypeScript-blue.svg)](https://www.typescriptlang.org/)
```
Comment (sbis04) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: This is using `JavaScript`, so you should update this tag to:
```
[![JavaScript](https://img.shields.io/badge/%3C%2F%3E-JavaScript-blue.svg)](https://www.javascript.com/)
```

**Comment by shwetalsoni** [comment]
@Rohitesh-Kumar-Jain and @proudofsimin since this PR has got 3 approvals, I request you to add it to merging list.
Thank you ;)

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+
+You can install wechaty-puppet as you like when using wechaty:
+
+1. If you want to use whatsapp, install wechaty-puppet-whatsapp
+
+```bash
+npm install wechaty-puppet-whatsapp
+```
+2. If you want to use wechat, install wechaty-puppet-wechat
+
```
Comment (lijiarui) [inline docusaurus/docs/examples/advanced/busy-bot.mdx:168]: I think you should change as @sbis04 request: https://github.com/wechaty/wechaty.js.org/pull/1099#discussion_r676183033

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+```bash
+npm install wechaty-puppet-wechat
+```
+
+3. If you want to use wecom, install wechaty-puppet-wxwork
```
Comment (lijiarui) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: This should refer: https://github.com/wechaty/wechaty.js.org/pull/1099#discussion_r676183045

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+npm install wechaty-puppet-wxwork
+```
+
+You can find more puppet information here: https://wechaty.js.org/docs/puppet-providers/
+
+Now, you are ready to write main code for bot.
+
+### 3. Writing code for bot
+
+Create a new folder called `src` and add a file `busy-bot.js`. You will be writing code here.
```
Comment (lijiarui) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: It seems you already create a folder `busy-bot`, so there is no need to create a new one `src` here.

File: docusaurus/docs/examples/advanced/busy-bot.mdx [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+
+Run the bot using the following command.
+
+```bash
+npx ts-node src/busy-bot.js
```
Comment (lijiarui) [inline docusaurus/docs/examples/advanced/busy-bot.mdx]: maybe we should change to `ts-node busy-bot.js` here.

**Review by lijiarui — CHANGES_REQUESTED** [review]

**Comment by shwetalsoni** [comment]
@lijiarui I have changed to requested changes.

**Review by lijiarui — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**